tests/unit/inference/test_checkpoint_sharding.py

```diff
@@ -69,7 +69,9 @@ class TestCheckpointShard(DistributedTest):
     world_size = 2
 
     def test(self, model_name, dtype, class_tmpdir, save_shard):
+        local_rank = int(os.getenv("LOCAL_RANK", "0"))
         world_size = int(os.getenv("WORLD_SIZE", "1"))
+
```

Comment:
Should this entire test be skipped if `dtype != torch.float16`, and thus eliminate the `if` condition in line 101?

Reply:
No, we still want to ensure we can properly load the checkpoints in `int8` (we verify this with `check_dtype`)